subsystems/drivesubsystem.py

```diff
@@ -53,9 +55,12 @@ def getState(self) -> SwerveModuleState:
 
     def applyState(self, state: SwerveModuleState) -> None:
         optimizedState = SwerveModuleState.optimize(state, self.getSwerveAngle())
```

Comment:
Is this call still necessary with the new optimization technique? I would have thought that the new algorithm would basically replace `SwerveModuleState.optimize()`

Reply:
@lmaxwell24 Sorry about the duplicate comments, let's continue the discussion here.
For the algorithm I would expect it to be hardware independent and of the form:
`betterOptimizedState = betterOptimize(state, self.getSwerveAngle(), minSwerveThreshold)`

Reply:
If part of the algorithm depends on `SwerveModuleState.optimize()` it can be called within `betterOptimize()`

Reply:
new optimization is angle specific optimization, new implementation is as such

Reply:
Looks like we're missing a call to "optimize" in the PWMSim case, otherwise looks good.

Reply:
not needed as optimization is inherited from the base `SwerveModule` class